src/stream/src/executor_v2/hop_window.rs

```diff
@@ -0,0 +1,369 @@
+            .ok_or_else(|| {
+                StreamExecutorError::InvalidArgument(format!(
+                    "window_size {} cannot be divided by window_slide {}",
+                    window_size, self.window_slide
+                ))
+            })?
+            .get();
+
+        let schema = self.info.schema;
+        let time_col_data_type = schema.fields()[time_col_idx].data_type();
+        let time_col_ref = InputRefExpression::new(time_col_data_type, self.time_col_idx).boxed();
+
+        let window_slide_expr =
+            LiteralExpression::new(DataType::Interval, Some(ScalarImpl::Interval(window_slide)))
+                .boxed();
+
+        // The first window_start of hop window should be:
+        // tumble_start(`time_col` - (`window_size` - `window_slide`), `window_slide`).
+        // Let's pre calculate (`window_size` - `window_slide`).
+        let window_size_sub_slide = window_size.checked_sub(&window_slide).ok_or_else(|| {
+            StreamExecutorError::InvalidArgument(format!(
+                "window_size {} cannot be subtracted by window_slide {}",
+                window_size, window_slide
+            ))
+        })?;
+        let window_size_sub_slide_expr = LiteralExpression::new(
+            DataType::Interval,
+            Some(ScalarImpl::Interval(window_size_sub_slide)),
+        )
+        .boxed();
+
+        let hop_start = new_binary_expr(
+            expr_node::Type::TumbleStart,
+            risingwave_common::types::DataType::Timestamp,
+            new_binary_expr(
+                expr_node::Type::Subtract,
+                DataType::Timestamp,
+                time_col_ref,
+                window_size_sub_slide_expr,
+            ),
+            window_slide_expr,
+        );
+
+        #[for_await]
+        for msg in input.execute() {
+            let msg = msg?;
+            let Message::Chunk(chunk) = msg else {
+                // TODO: syn has not supported `let_else`, we desugar here manually.
+                yield std::task::Poll::Ready(msg);
+                continue;
+            };
+            // TODO: compact may be not necessary here.
+            let chunk = chunk.compact().map_err(StreamExecutorError::ExecutorV1)?;
+            let (data_chunk, ops) = chunk.into_parts();
+            let hop_start = hop_start
+                .eval(&data_chunk)
+                .map_err(StreamExecutorError::EvalError)?;
+            let hop_start_chunk = DataChunk::new(vec![Column::new(hop_start)], None);
+            let (origin_cols, visibility) = data_chunk.into_parts();
+            // SAFETY: Already compacted.
+            assert!(visibility.is_none());
+            for i in 0..units {
+                let window_start_offset =
+                    self.window_slide.checked_mul_int(i).ok_or_else(|| {
+                        StreamExecutorError::InvalidArgument(format!(
+                            "window_slide {} cannot be multiplied by {}",
```

Comment:
Just curious... How can this happen?

Reply:
Not sure :(

Reply:
The `IntervalUnit` can only represent months in u32，if the month is too big or the units are too big, it will happened (

Reply:
Same to days